src/dex/gmx/gmx.ts

```diff
@@ -0,0 +1,275 @@
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (side === SwapSide.BUY || !this.pool) return [];
+    const srcAddress = wrapETH(srcToken, this.network).address.toLowerCase();
+    const destAddress = wrapETH(destToken, this.network).address.toLowerCase();
+    if (
+      this.supportedTokensMap[srcAddress] &&
+      this.supportedTokensMap[destAddress]
+    ) {
+      return [`${this.dexKey}_${srcAddress}`, `${this.dexKey}_${destAddress}`];
+    }
+    return [];
+  }
+
+  // Returns pool prices for amounts.
+  // If limitPools is defined only pools in limitPools
+  // should be used. If limitPools is undefined then
+  // any pools can be used.
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<GMXData>> {
+    if (side === SwapSide.BUY || !this.pool) return null;
+    const srcAddress = wrapETH(srcToken, this.network).address.toLowerCase();
+    const destAddress = wrapETH(destToken, this.network).address.toLowerCase();
+    if (
+      !(
+        this.supportedTokensMap[srcAddress] &&
+        this.supportedTokensMap[destAddress]
+      )
+    )
+      return null;
+    const srcPoolIdentifier = `${this.dexKey}_${srcAddress}`;
+    const destPoolIdentifier = `${this.dexKey}_${destAddress}`;
+    const pools = [srcPoolIdentifier, destPoolIdentifier];
+    if (limitPools && pools.some(p => !limitPools.includes(p))) return null;
+
+    const unitVolume = getBigIntPow(srcToken.decimals);
+    const prices = await this.pool.getAmountOut(
+      srcAddress,
+      destAddress,
+      [unitVolume, ...amounts],
+      blockNumber,
+    );
+
+    if (!prices) return null;
+
+    return [
+      {
+        prices: prices.slice(1),
+        unit: prices[0],
+        gasCost: 0, // TODO: fix gas cost
+        exchange: this.dexKey,
+        data: {},
+        poolAddresses: [this.params.vault],
+      },
+    ];
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: GMXData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    return {
+      targetExchange: this.params.vault,
+      payload: '0x',
+      networkFee: '0',
+    };
+  }
+
+  getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: GMXData,
+    side: SwapSide,
+  ): SimpleExchangeParam {
+    return {
+      callees: [srcToken, this.params.vault],
+      calldata: [
+        GMX.erc20Interface.encodeFunctionData('transfer', [
+          this.params.vault,
+          srcAmount,
+        ]),
+        Vault.interface.encodeFunctionData('swap', [
+          srcToken,
+          destToken,
+          this.augustusAddress,
+        ]),
+      ],
+      values: ['0', '0'],
+      networkFee: '0',
+    };
+  }
+
+  async updatePoolState(): Promise<void> {
```

Comment:
I don't think pool tracker or pools helper actually calls this, so it's not going to work.

Reply:
I think it would be better to fix the pool helper to allow for this. We can leave this as it is and fix the pools helper as a separate task